### PR TITLE
Add support for empty `request.json()`.

### PR DESCRIPTION
**Pull request by olivertso**
The app throws a `json.decoder.JSONDecodeError` if I try to access `request.json()` when the request has an empty body.

Hope this is helpful :)

### CODE REVIEW

**Comment by lovelydinosaur** [comment]
Thanks! It's not obvious that this is the behavior that we want.
I'm inclined to think that we should error with 400 if we expected JSON content, but there is an empty body.

File: starlette/requests.py [hunk 1]
```diff
@@ -103,6 +103,8 @@ def cookies(self) -> typing.Dict[str, str]:
     async def json(self) -> typing.Any:
         if not hasattr(self, "_json"):
             body = await self.body()
+            if not body:
+                body = b"{}"
```
Comment (alexbotello) [inline starlette/requests.py:107]: You could probably do something like:
```suggestion
                if "app" in self._scope:
                    raise HTTPException(status_code=400)
                return PlainTextResponse("Bad Request", status_code=400)
```

**Comment by alexbotello** [comment]
@olivertso you still up for making this change? I’d actually really like to get this behavior in. If not, i’ll go ahead and submit a PR

**Comment by olivertso** [comment]
@alexbotello Sorry for the delay. I'm still up but I'm really busy lately. Please submit your PR. Thanks.

**Pull request closed without merging**